dts/bindings/pinctrl/raspberrypi,pico-pinctrl.yaml

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2021 Yonatan Schachter
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    RaspberryPi Pico Pin controller Node
+    Based on pincfg-node.yaml binding.
+
+compatible: "raspberrypi,pico-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        child-binding:
+          property-allowlist:
+            - bias-disable
+            - bias-pull-down
+            - bias-pull-up
+            - input-enable
+            - input-schmitt-enable
+
+child-binding:
+  description: |
+    Definitions for a pinctrl state.
+  child-binding:
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          An array of pins sharing the same group properties. Each
+          element of the array is an integer constructed from the
+          pin number and the alternative function of the pin.
+      drive-strength:
+        required: false
+        type: int
+        enum:
+          - 2
+          - 4
+          - 8
+          - 12
+      slew-rate-fast:
+        required: false
+        type: boolean
```

Comment:
you can use standard `slew-rate` property here (with custom enum maybe), check e.g. gigadevice or ST